web-app/src/common/graphWinPrediction.ts

```diff
@@ -0,0 +1,131 @@
+import * as jkstra from 'jkstra';
+import {
+  doesCellExist, doesCellExistAndHaveStone, GameState, StoneColor,
+  BLACK_NODE_START, BLACK_NODE_END, WHITE_NODE_START, WHITE_NODE_END
+} from './gameState';
+import { Coordinates } from './utils';
+
+export const UNDEFINED_EDGE_COST = -1;
+export const LOW_EDGE_COST = 1;
+export const HIGH_EDGE_COST = 100;
+
+export interface HexBoardWinPredictionGraph {
+  winPrediction: jkstra.Graph;
+  vertexMap: Map<string, jkstra.Vertex>;
+}
```

Comment:
I don't understand why you need both this interface and this one:

```tx
export interface HexBoardWinDetectionGraph {
  winDetectionGraph: jkstra.Graph;
  vertexMap: Map<string, jkstra.Vertex>;
}
```

Can't you factorize?

Moreover, I found the code to be clearer when the first attribute was called `graph`. Here, in your code, you have things like `winPrediction.addVertex(id)` which I find less readable than  `graph.addVertex(id)` .

Reply:
I voluntarily chose to split it in 2 differents interfaces to be explicit about the diff between the 2 kinds of graph and avoid using them in the wrong context.
The use of a specific name for the graph property(like winPrediction instead of graph) is to avoid the [duck-typing](https://www.javatpoint.com/typescript-duck-typing) TypeScript feature ! But we could use another name, as long as they are differents in the 2 interfaces

Reply:
I still don't agree fully, since I don't really see any place where we could confuse `HexBoardWinDetectionGraph` with `HexBoardWinPredictionGraph` (in the end, in `pathfinding.ts`, we just export 2 functions with different signatures and names, so there is no risk of confusion, and the graphs are only used internally for calculations).

But on the other hand, I agree that this is not really important. At least, for now. These two interfaces can co-exist, and I don't think it will cause huge maintenance problems later either. Let's leave it that way.